contracts/cw721-base/src/contract.rs

```diff
@@ -450,15 +463,22 @@ fn query_all_approvals<S: Storage, A: Api, Q: Querier>(
 
     let res: StdResult<Vec<_>> = operators_read(&deps.storage, &owner_raw)
         .range(start.as_deref(), None, Order::Ascending)
-        .take(limit)
-        .map(|item| {
-            item.and_then(|(k, expires)| {
-                Ok(cw721::Approval {
-                    spender: deps.api.human_address(&k.into())?,
-                    expires,
-                })
-            })
+        .filter_map(|item| match item {
+            Ok((k, expires)) => {
+                // we remove all expired operators from the result
```

Comment:
Sure about this? I'm not sure, but listing expired operators can be useful for debugging / accountability issues.

Reply:
I guess I can keep them, but then add another flag to the response "is_expired: bool". Is it useful info on the client, without forcing calculations there

Reply:
Good idea. Or directly add the expiration info.

Another option would be to add a boolean flag to the query, i. e. `include_expired` or so.

Reply:
Should be addressed here: https://github.com/CosmWasm/cosmwasm-plus/pull/111/commits/09b7b31427eb9c484e070697c91dba89c1a22310

Reply:
> Another option would be to add a boolean flag to the query, i. e. include_expired or so.

Just read this. I actually prefer this idea

Reply:
I will undo my change and redo it the other way. Let me know which commit you prefer.

Reply:
I too think the query option is better.

Reply:
Look at https://github.com/CosmWasm/cosmwasm-plus/pull/111/commits/0525a341b7bc29cddaa40405c84ad4561aef6029